gitalias.txt

```diff
@@ -156,11 +156,109 @@
 
   # checkout - update the working tree to match a branch or paths. [same as "o" for "out"]
   co = checkout
+  cog = checkout -guess
   cong = checkout --no-guess
 
   # create a new branch from the current working tree, and switch to it
   cob = checkout -b
 
+  # cof - switch to a branch's first commit hash.
+  #
+  # Syntax:
+  #
+  #    git cof [branch name]
+  #
+  # Options:
+  #
+  #   * branch name: default is the current branch name.
+  #
+  # Example: switch to the current branch's first commit hash:
+  #
+  #     git cof
+  #
+  # Example: switch to the "foo" branch's first commit hash:
+  #
+  #     git cof foo
+  #
+  cof = !"f() { \
+    REMOTE=\"$(git remote)\"; \
+    branch=\"${1:-$(git symbolic-ref \"refs/remotes/$REMOTE/HEAD\")}\"; \
```

Comment:
And if there are no parameters why work with remotes at all? In detached HEAD state remote `HEAD` could be completely different from the current `HEAD`, it's hardly a way to get the current branch. There is no reliable way to get the current branch in detached HEAD state. If my current branch is `development` and current commit is `development~10`, but the remote `HEAD` is `refs/remotes/origin/master` then what?

IMO the best course of actions would be to get the current branch from the `HEAD` and in case of detached HEAD fail and force user to provide branch name.

Reply:
Or may be guess with `git branch --contains HEAD` ?

Reply:
`git for-each-ref --contains HEAD --count=1 --format="%(refname)" refs/heads/`